features/qskconfig.cmake

```diff
@@ -0,0 +1,169 @@
+############################################################################
+# QSkinny - Copyright (C) 2016 Uwe Rathmann
+# This file may be used under the terms of the QSkinny License, Version 1.0
+############################################################################
+
+set(QSK_VERSION ${CMAKE_PROJECT_VERSION})
+
+# trying the PREFIX environment variable first
+set(QSK_INSTALL_PREFIX $ENV{PREFIX}) # TODO still required?
+
+if("${QSK_INSTALL_PREFIX}" STREQUAL "")
+    set(QSK_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}") # TODO where does QT_INSTALL_PREFIX come from?
+
+    if (UNIX)
+        set(QSK_INSTALL_PREFIX /usr/local/qskinny-${QSK_VERSION})
+    endif()
+
+    if (MSVC)
+        set(QSK_INSTALL_PREFIX C:/Qskinny-${QSK_VERSION})
+    endif()
+endif()
+
+# TODO find compiler flag equivalent
+list(APPEND CONFIG        no_private_qt_headers_warning)
+list(APPEND CONFIG        warn_on)
+list(APPEND CONFIG        no_keywords)
+list(APPEND CONFIG        silent)
+#list(APPEND CONFIG           -= depend_includepath) # TODO was -=
+list(APPEND CONFIG        strict_c++)
+list(APPEND CONFIG        c++11)
+list(APPEND CONFIG        pedantic)
+list(APPEND CONFIG        hide_symbols)
+#list(APPEND CONFIG           += debug)
+#list(APPEND CONFIG           += sanitize)
+
+# TODO
+# use_no_rpath {
+#     CONFIG -= use_local_rpath use_install_rpath
+# } else {
+#     cross_compile {
+#         CONFIG *= use_install_rpath
+#         QMAKE_RPATHLINKDIR *= $${QSK_PLUGIN_DIR}/skins
+#     } else {
+#         !use_install_rpath: CONFIG *= use_local_rpath
+#     }
+# }
+
+set(MOC_DIR      moc)
+set(OBJECTS_DIR  obj)
+set(RCC_DIR      rcc)
+
+option(BUILD_QSKDLL "build qskinny as shared library" ON)
```

Comment:
@uwerat currently it is either build qskinny as a shared library (as well as the skins and support library) or as static libraries. 

In the future we could also add the option for building both static and shared library, but for now this should be sufficient.